presto-iceberg/src/test/java/com/facebook/presto/iceberg/hive/TestIcebergHiveStatistics.java

```diff
@@ -402,6 +408,41 @@ public void testPredicateOnlyColumnInStatisticsOutput(boolean pushdownFilterEnab
         }
     }
 
+    @Test
+    public void testStatisticsCachePartialEviction()
+            throws Exception
+    {
+        try (DistributedQueryRunner queryRunner = createIcebergQueryRunner(ImmutableMap.of(), ImmutableMap.of("iceberg.max-statistics-file-cache-size", "1024B"))) {
+            Session session = Session.builder(queryRunner.getDefaultSession())
+                    // set histograms enabled
+                    .setSystemProperty(OPTIMIZER_USE_HISTOGRAMS, "true")
+                    .setCatalogSessionProperty("iceberg", STATISTICS_KLL_SKETCH_K_PARAMETER, "32768")
+                    .build();
+
+            queryRunner.execute(session, "ANALYZE lineitem");
+            // get table statistics, to populate some of the cache
+            TableStatistics statistics = getTableStatistics(queryRunner, session, "lineitem");
+            RuntimeStats runtimeStats = session.getRuntimeStats();
+            runtimeStats.getMetrics().keySet().stream().filter(name -> name.contains("ColumnCount")).findFirst()
+                    .ifPresent(stat -> assertEquals(32, runtimeStats.getMetric(stat).getSum()));
+            runtimeStats.getMetrics().keySet().stream().filter(name -> name.contains("PuffinFileSize")).findFirst()
+                    .ifPresent(stat -> assertTrue(runtimeStats.getMetric(stat).getSum() > 1024));
+            // get them again to trigger retrieval of _some_ cached statistics
```

Comment:
Can we assert that eviction count was more than 0 ? This proves that the cache was maxed out and not all read statistics have been cached